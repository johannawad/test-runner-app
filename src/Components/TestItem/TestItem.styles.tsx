import styled from 'styled-components';

export const Card = styled.div`
  display: flex;
  flex-direction: column;
  border: 1px solid rgb(77, 77, 34);
  width: 400px;
  margin: 10px;
  padding: 10px;
`;

export const TextBold = styled.p`
    color: ${props => props.color? props.color: 'white'};
    font-size: 16px;
    font-weight: 600;
`;