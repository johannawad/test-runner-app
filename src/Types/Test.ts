export interface Test {
  id?: number;
  description: string;
  status?: TestStatus;
  run(callback: any): void;
}

export enum TestStatus {
  NotStarted = 'Not Started',
  Running = 'Running',
  Passed = 'Passed',
  Failed = 'Failed'
}
