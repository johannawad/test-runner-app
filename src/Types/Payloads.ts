import { TestStatus } from './Test';

export interface TestStatusUpdatePayload {
  id: number;
  status: TestStatus;
}
