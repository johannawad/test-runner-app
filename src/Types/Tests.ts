import { Test } from './Test';

export interface TestsState {
  data: Test[];
  loading: boolean;
  errors?: string;
}

export enum TestsActionTypes {
  FETCH_REQUEST = '@@tests/FETCH_REQUEST',
  FETCH_SUCCESS = '@@tests/FETCH_SUCCESS',
  FETCH_ERROR = '@@tests/FETCH_ERROR',
  RUN_ALL = '@@tests/RUN_ALL',
  UPDATE_TEST_STATUS = '@@tests/RUN_TEST'
}
