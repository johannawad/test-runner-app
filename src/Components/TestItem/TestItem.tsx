import React from 'react';
import { Test, TestStatus } from '../../Types/Test';
import TextUtils from '../../Utils/text.utils';
import { Card, TextBold } from './TestItem.styles';

interface TestComponentProps {
  test: Test;
}

const TestItem: React.FC<TestComponentProps> = ({
  test
}: TestComponentProps) => {
  const textColor = test.status === TestStatus.Passed? 'green': test.status === TestStatus.Failed ? 'red': 'white';
  return (
    <Card>
      <h3>
        {Number(test.id) + 1}:{' '}
        {TextUtils.capitalizeFirstLetter(test.description)}
      </h3>
      <br />
      <p>
        Status: 
      </p>
      <TextBold color={textColor}>{test.status}</TextBold>
    </Card>
  );
};

export { TestItem };
