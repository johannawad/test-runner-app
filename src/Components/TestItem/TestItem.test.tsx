import React from "react";
import { TestItem } from "./TestItem";
import { shallow } from "enzyme";
import { Test, TestStatus } from "../../Types/Test";
describe("TestItem", () => {
  it("should render correctly", () => {
    const test: Test = {
      description: "desc",
      run: jest.fn,
      status: TestStatus.Failed,
    };
    const component = shallow(<TestItem test={test} />);
    expect(component).toMatchSnapshot();
  });

  it.each([
    [TestStatus.Failed, "red"],
    [TestStatus.Passed, "green"],
    [TestStatus.Running, "white"],
    [TestStatus.NotStarted, "white"],
  ])("should when status is %s be of color %s ", (status, color) => {
    const test: Test = { description: "desc", run: jest.fn, status };
    const component = shallow(<TestItem test={test} />);
    const text = component.find({ children: status }).get(0);
    expect(text.props.color).toBe(color);
  });
});
