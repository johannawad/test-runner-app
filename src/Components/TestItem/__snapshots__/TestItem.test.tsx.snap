// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestItem should render correctly 1`] = `
<styled.div>
  <h3>
    NaN
    :
     
    Desc
  </h3>
  <br />
  <p>
    Status:
  </p>
  <styled.p
    color="red"
    data-testId="text-bold"
  >
    Failed
  </styled.p>
</styled.div>
`;
